### ISSUE TITLE
Numeric underflow when loss gets small.

### ISSUE THREAD

**Issue by dhruvrajan** [dhruvrajan/tensorflow-keras-java#13]
When training MNIST for > 14 epochs, there seems to be an underflow problem, resulting in NaNs when the loss becomes ~ 7E-3.
